File: ios/SpentyAI/Core/Models/Category.swift
import Foundation

enum CategoryType: String, Codable, Hashable, CaseIterable {
    case income
    case expense
}

struct Category: Codable, Identifiable, Hashable {
    var id: String { categoryId }

    let categoryId: String
    let userId: String?
    var name: String
    var categoryType: CategoryType
    var parentId: String?
    var createdAt: String?

    enum CodingKeys: String, CodingKey {
        case categoryId = "category_id"
        case userId = "user_id"
        case name
        case categoryType = "category_type"
        case parentId = "parent_id"
        case createdAt = "created_at"
    }
}

struct CategoryCreate: Codable {
    var name: String
    var categoryType: CategoryType
    var parentId: String?

    enum CodingKeys: String, CodingKey {
        case name
        case categoryType = "category_type"
        case parentId = "parent_id"
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/models/Category.kt
package com.spentyai.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Category(
    @SerialName("categoryId") val id: String = "",
    val name: String? = null,
    @SerialName("categoryType") val categoryType: String? = null,
    @SerialName("parentId") val parentId: String? = null,
    var children: List<Category>? = null
)

enum class CategoryType(val value: String) {
    INCOME("income"),
    EXPENSE("expense");

    companion object {
        fun fromString(value: String): CategoryType =
            entries.firstOrNull { it.value == value.lowercase() } ?: EXPENSE
    }
}

@Serializable
data class CreateCategoryRequest(
    val name: String,
    @SerialName("categoryType") val categoryType: String,
    @SerialName("parentId") val parentId: String? = null
)

@Serializable
data class UpdateCategoryRequest(
    val name: String
)

@Serializable
data class MergeCategoriesRequest(
    @SerialName("sourceId") val sourceId: String,
    @SerialName("targetId") val targetId: String
)

@Serializable
data class MergeCategoriesResponse(
    val success: Boolean = false,
    val message: String? = null
)
